migration/sql-files/012-add-state-to-auth-state-reference.sql

```diff
@@ -0,0 +1,2 @@
+-- Alter Oauth state reference table to add state used in oauth workflow
+ALTER TABLE oauth_state_references ADD COLUMN state TEXT unique;
```

Comment:
To enforce a not null, non-zero-length string, I believe this is the correct DDL:

ALTER TABLE oauth_state_references ADD COLUMN state TEXT unique NOT NULL CHECK (oauth_state_references <> '')

Reply:
I will let @sbryzak to help you with correct SQL statement here.
We need to make sure:
1. State can be not empty unique string (in case it's not optimal to check if the string is empty on the DB level we could check it in our Create() method in the repo.
2. We have an unique index for the state column which is used when we look up the state.

Reply:
After testing against my local postgres (v9.6.3) db, I can confirm the above DDL works.  Creating a unique column causes the db to create a corresponding index to enforce the unique constraint:

```
postgres=# ALTER TABLE oauth_state_references ADD COLUMN state TEXT unique NOT NULL CHECK (state <> '');
ALTER TABLE
postgres=# \d oauth_state_references
                       Table "public.oauth_state_references"
   Column   |           Type           | Collation | Nullable |      Default
------------+--------------------------+-----------+----------+--------------------
 created_at | timestamp with time zone |           |          | 
 updated_at | timestamp with time zone |           |          | 
 deleted_at | timestamp with time zone |           |          | 
 id         | uuid                     |           | not null | uuid_generate_v4()
 referrer   | text                     |           | not null | 
 state      | text                     |           | not null | 
Indexes:
    "oauth_state_references_pkey" PRIMARY KEY, btree (id)
    "oauth_state_references_state_key" UNIQUE CONSTRAINT, btree (state)
Check constraints:
    "oauth_state_references_state_check" CHECK (state <> ''::text)
postgres=#